rust-toolchain.toml

```diff
@@ -1,3 +1,3 @@
 [toolchain]
-channel = "1.55.0"
+channel = "1.58.0"
```

Comment:
We may need to wait either until common is updated from 1.55, or the sdk release process doesn't use nix, because right now agent-rs needs to be compatible with toolchain 1.55.